src/openzaak/components/zaken/signals.py

```diff
@@ -53,3 +62,69 @@ def sync_zaakbesluit(
 
     else:
         raise NotImplementedError(f"Signal {signal} is not supported")
+
+
+def sync_create_zaakcontactmoment(relation: ZaakContactMoment):
+    zaak_url = build_absolute_url(
+        reverse("zaak-detail", kwargs={"uuid": relation.zaak.uuid})
+    )
+
+    logger.info("Zaak: %s", zaak_url)
+    logger.info("Contactmoment: %s", relation.contactmoment)
+
+    # Define the remote resource with which we need to interact
+    resource = "objectcontactmoment"
+    client = Service.get_client(relation.contactmoment)
+
+    try:
+        response = client.create(
+            resource,
+            {
+                "object": zaak_url,
+                "contactmoment": relation.contactmoment,
+                "objectType": "zaak",
+            },
+        )
+    except Exception as exc:
+        logger.error("Could not create remote relation", exc_info=1)
+        raise SyncError("Could not create remote relation") from exc
+
+    # save ZaakBesluit url for delete signal
+    relation._objectcontactmoment = response["url"]
+    relation.save()
+
+
+def sync_delete_zaakcontactmoment(relation: ZaakContactMoment):
+    resource = "objectinformatieobject"
+    client = Service.get_client(relation.contactmoment)
+
+    try:
+        client.delete(resource, url=relation._objectcontactmoment)
+    except Exception as exc:
+        logger.error("Could not delete remote relation", exc_info=1)
+        raise SyncError("Could not delete remote relation") from exc
+
+
+@receiver(
+    [post_save, pre_delete],
+    sender=ZaakContactMoment,
+    dispatch_uid="sync.sync_contactmoment_relation",
+)
+def sync_contactmoment_relation(sender, instance: ZaakContactMoment = None, **kwargs):
+    signal = kwargs["signal"]
+    if signal is post_save and not instance._objectcontactmoment:
+        sync_create_zaakcontactmoment(instance)
+    elif signal is pre_delete and instance._objectcontactmoment:
+        cache = caches["kic_sync"]
+        marked_zcms = cache.get("zcms_marked_for_delete")
+        if marked_zcms:
+            cache.set("zcms_marked_for_delete", marked_zcms + [instance.uuid])
+        else:
+            cache.set("zcms_marked_for_delete", [instance.uuid])
+
+        try:
+            sync_delete_zaakcontactmoment(instance)
+        finally:
+            marked_zcms = cache.get("zcms_marked_for_delete")
+            marked_zcms.remove(instance.uuid)
+            cache.set("zcms_marked_for_delete", marked_zcms)
```

Comment:
This seems just copied over from the reference implementation instead of applying the same patterns from Open Zaak to deal with these structures.